src/components/ui/chart/stacked-bar-chart/stacked-bar-chart2.tsx

```diff
@@ -0,0 +1,349 @@
+        flexDirection: "column",
+        gap: "8px",
+      },
+      title: {
+        color: themePalette.tertiary,
+        fontSize: "12px",
+        fontWeight: "400",
+      },
+      itemContainer: {
+        display: "flex",
+        flexDirection: "column",
+        gap: "2px",
+      },
+      itemRow: {
+        display: "flex",
+        alignItems: "center",
+        gap: "44px",
+      },
+      labelContainer: {
+        display: "flex",
+        alignItems: "center",
+        gap: "4px",
+      },
+      colorDot: {
+        width: "8px",
+        height: "8px",
+        borderRadius: "50px",
+      },
+      labelText: {
+        color: themePalette.primary,
+        fontSize: "12px",
+        fontWeight: "400",
+      },
+      valueContainer: {
+        display: "flex",
+        alignItems: "flex-end",
+        justifyContent: "center",
+        flexDirection: "column",
+        flex: "1",
+        gap: "1px",
+        lineHeight: "16px",
+      },
+      valueText: {
+        color: themePalette.primary,
+        fontSize: "11px",
+      },
+      integerPart: {
+        color: themePalette.primary,
+        fontSize: "11px",
+        fontWeight: "600",
+      },
+      decimalPart: {
+        color: themePalette.primary,
+        fontSize: "10px",
+      },
+      denomText: {
+        color: themePalette.primary,
+        fontSize: "9px",
+        marginLeft: "2px",
+      },
+    }),
+    [themePalette],
+  );
+
+  const styleToString = React.useCallback((styleObj: Record<string, string | number>) => {
+    return Object.entries(styleObj)
+      .map(([key, value]) => {
+        const cssKey = key.replace(/([A-Z])/g, "-$1").toLowerCase();
+        return `${cssKey}: ${value}`;
+      })
+      .join("; ");
+  }, []);
+
+  const formatAmountText = React.useCallback(
+    (value: number | string | undefined | null, decimals = 6): { integer: string; decimal: string } => {
+      if (value === undefined || value === null) {
+        return { integer: "0", decimal: "" };
+      }
+
+      const valueStr = typeof value === "string" ? value.replace(/,/g, "") : value.toString();
+      if (new BigNumber(valueStr).isNaN() || valueStr.length === 0) {
+        return { integer: "0", decimal: "" };
+      }
+
+      const numbers = valueStr.split(".");
+      const integer = new BigNumber(numbers[0]).toFormat(0);
+
+      if (numbers.length > 1 && numbers[1] !== "0") {
+        const decimal = numbers[1].slice(0, decimals);
+        return { integer, decimal: `.${decimal}` };
+      }
+
+      return { integer, decimal: "" };
+    },
+    [],
+  );
+
+  const renderTooltipItem = React.useCallback(
+    (param: echarts.DefaultLabelFormatterCallbackParams) => {
+      const backgroundColor = param.color ? param.color.toString() : themePalette.primary;
+      const colorDotStyle = { ...tooltipInlineStyles.colorDot, background: backgroundColor };
+
+      const value = param.value as number | string | null | undefined;
```

Comment:
this type narrowing is definitely unavoidable. However, using `as` for type assertion can compromise runtime safety. It would be better to keep the value as `unknown` and use type guards for safe handling.

something like (pseudo code):
```typescript
const extractSafeParamValue = (value: unknown): number | string | null | undefined => {
  if (isValidParamValue(value)) {
    return value;
  }
  return null;
};
```